eox_core/api/v1/tests/integration/test_views.py

```diff
@@ -0,0 +1,384 @@
+"""
+Integration test suite for the API v1 views.
+"""
+
+from __future__ import annotations
+
+import requests
+from ddt import data as ddt_data
+from ddt import ddt, unpack
+from django.conf import settings
+from django.test import TestCase
+from django.urls import reverse
+from faker import Faker
+from rest_framework import status
+
+fake = Faker()
+
+
+API_TIMEOUT = 5
+CLIENT_ID = "client_id"
+CLIENT_SECRET = "client_secret"
+FAKE_USERS_DATA = iter(
+    [
+        {
+            "username": fake.unique.user_name(),
+            "email": fake.unique.email(),
+            "fullname": fake.unique.name(),
+            "password": fake.unique.password(),
+            "activate_user": True,
+        }
+        for _ in range(100)
+    ]
+)
+
+
+class BaseAPIIntegrationTest(TestCase):
+    """
+    Base class for the integration test suite.
+    """
+
+    def setUp(self):
+        """
+        Set up the test suite.
+        """
+        self.default_site = self.get_tenant_data()
+        self.tenant_x = self.get_tenant_data()
+
+    def get_tenant_data(self, prefix: str = "") -> dict:
+        """
+        Get the tenant data.
+
+        If no prefix is provided, the default site data is returned.
+
+        Args:
+            prefix (str): The tenant prefix.
+
+        Returns:
+            dict: The tenant data.
+        """
+        domain = f"{prefix}.{settings.LMS_BASE}" if prefix else settings.LMS_BASE
+        return {
+            "base_url": f"http://{domain}",
+            "domain": domain,
+        }
+
+    def get_access_token(self, tenant_base_url: str) -> str:
+        """
+        Get an access token for a tenant.
+
+        Args:
+            tenant_base_url (str): The tenant base URL.
+
+        Returns:
+            str: The access token.
+        """
+        data = {
+            "client_id": CLIENT_ID,
+            "client_secret": CLIENT_SECRET,
+            "grant_type": "client_credentials",
+        }
+        url = f"{tenant_base_url}/oauth2/access_token/"
+        response = requests.post(url, data=data, timeout=API_TIMEOUT)
+        print(f"\n\nResponse: {response}, {response.text}\n\n")
+        return response.json()["access_token"]
+
+    # pylint: disable=too-many-arguments
+    def make_tenant_request(
+        self,
+        tenant: dict,
+        method: str,
+        url: str,
+        data: dict | None = None,
+        params: dict | None = None,
+    ) -> requests.Response:
```

Comment:
I'm sorry, I forgot to leave this comment. 

This method assumes we're working with tenants, but that's not always the case. Should we call it `make_request` instead?

Reply:
Yes, you're right. I updated the method name.